### PR TITLE
Feat: Enables media queries & mixins overrides

### PR DESCRIPTION
**Pull request by renatamottam**
## What's the purpose of this pull request?
- [x] Enables mixins (breakpoints) override;
- [x] [Doc] Explain via doc how to override mixins;
- [x] [Doc] Explain about breakpoints mixins;

## How to test it?
`Documentation: Review modified pages`
- [Global Tokens > Grid and Layout](https://faststore-site-git-feat-media-queries-fs-1207-faststore.vercel.app/docs/global-tokens/grid-and-layout)
- [Mixins & Breakpoints](https://faststore-site-git-feat-media-queries-fs-1207-faststore.vercel.app/docs/customizing/mixins-breakpoints)

`Breakpoints and mixins override`
- Follow the steps listed on [Mixins & Breakpoints](https://faststore-site-git-feat-media-queries-fs-1207-faststore.vercel.app/docs/customizing/mixins-breakpoints) on the https://github.com/vtex-sites/starter.store/pull/134

## Starters Deploy Preview
- [Feat: Enables media queries & mixins overrides](https://github.com/vtex-sites/starter.store/pull/134)

## Related Issue
#1905

### CODE REVIEW

File: apps/site/pages/docs/customizing/mixins-breakpoints.mdx [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+<header>
+
+# Mixins & Breakpoints
+
+</header>
+
+
+## Breakpoints
+
+In order to help our breakpoint customization, we use [Include-Media](https://eduardoboucas.github.io/include-media/#features), a Sass library for writing CSS media queries in an easy and maintainable way, using a natural and simplistic syntax. We use 5 basic breakpoints:
```
Comment (hellofanny) [inline apps/site/pages/docs/customizing/mixins-breakpoints.mdx:18]:
```suggestion
To ease our screen breakpoint customization, we're using [Include-Media](https://eduardoboucas.github.io/include-media/#features), a Sass library for writing CSS media queries in an easy and maintainable way, using a natural and simplistic syntax. We've defined the 5 following screen breakpoints:
```

File: apps/site/pages/docs/customizing/mixins-breakpoints.mdx [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+  "phonemid":      375px,
+  "tablet":        768px,
+  "notebook":      1280px,
+  "desktop":       1440px,
+);
+```
+
+### Custom Breakpoints
+
+If you want to override these sizing system, just follow these steps:
```
Comment (hellofanny) [inline apps/site/pages/docs/customizing/mixins-breakpoints.mdx:32]:
```suggestion
If you want to override this sizing system, follow the following steps:
```

File: apps/site/pages/docs/customizing/mixins-breakpoints.mdx [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+
+1. Go to the `/src` folder and create the `styles` folder and inside it. You can run the following command to create them:
```
Comment (hellofanny) [inline apps/site/pages/docs/customizing/mixins-breakpoints.mdx:34]:
```suggestion
1. Go to the `src` folder and create the `styles` folder inside it. You can run the following command to create it:
```

File: apps/site/pages/docs/customizing/mixins-breakpoints.mdx [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+$breakpoints: (
+  "phone":         320px,
+  "phonemid":      375px,
+  "tablet":        768px,
+  "notebook":      1280px,
+  "desktop":       1440px,
+);
+```
+
+### Custom Breakpoints
```
Comment (hellofanny) [inline apps/site/pages/docs/customizing/mixins-breakpoints.mdx:30]:
```suggestion
### Customizing Breakpoints
```

File: apps/site/pages/docs/customizing/mixins-breakpoints.mdx [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+# Mixins & Breakpoints
+
+</header>
+
+
+## Breakpoints
+
+In order to help our breakpoint customization, we use [Include-Media](https://eduardoboucas.github.io/include-media/#features), a Sass library for writing CSS media queries in an easy and maintainable way, using a natural and simplistic syntax. We use 5 basic breakpoints:
+
+```scss copy
```
Comment (hellofanny) [inline apps/site/pages/docs/customizing/mixins-breakpoints.mdx:20]:
```suggestion
```scss
```

File: apps/site/pages/docs/customizing/mixins-breakpoints.mdx [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+  "phone":         320px,
+  "phonemid":      375px,
+  "tablet":        768px,
+  "notebook":      1280px,
+  "desktop":       1500px,
+);
+```
+
+<Callout type="warning" emoji="⚠️">
+  You need to declare all the variants (phone, phonemid, tablet, notebook and desktop).
```
Comment (hellofanny) [inline apps/site/pages/docs/customizing/mixins-breakpoints.mdx:79]:
```suggestion
  You need to declare all the variants (phone, phonemid, tablet, notebook and desktop) even when changing only one of the values.
```

File: apps/site/pages/docs/customizing/mixins-breakpoints.mdx [hunk 7]
```diff
@@ -0,0 +1,246 @@
...
+</Callout>
+
+4. Run `yarn dev` in the terminal to start the development server.
```
Comment (hellofanny) [inline apps/site/pages/docs/customizing/mixins-breakpoints.mdx:82]:
```suggestion
4. Run `yarn dev` in the terminal to start the development server. Your should be able to see the changes.
```

File: apps/site/pages/docs/customizing/mixins-breakpoints.mdx [hunk 8]
```diff
@@ -0,0 +1,246 @@
...
+
```
Comment (hellofanny) [inline apps/site/pages/docs/customizing/mixins-breakpoints.mdx:83]: We should add something like:

```suggestion
Modifying these breakpoints implies changes to certain global tokens. See [Grid & Layout](/docs/global-tokens/grid-and-layout#breakpoints) for more details.
```

File: apps/site/pages/docs/customizing/mixins-breakpoints.mdx [hunk 9]
```diff
@@ -0,0 +1,246 @@
...
+---
+
+## Mixins
+
+We use the [Sass `@mixin`](https://sass-lang.com/documentation/at-rules/mixin/) to allow you to define styles that can be re-used throughout your stylesheet. We divide them into these categories:
```
Comment (hellofanny) [inline apps/site/pages/docs/customizing/mixins-breakpoints.mdx:88]:
```suggestion
We use the [Sass `@mixin`](https://sass-lang.com/documentation/at-rules/mixin/) to allow you to define styles that can be re-used throughout your application. We divide them into 3 categories: `Global Mixins`, `Layout & Spacing Mixins` and `Focus Ring Mixins`.
```

File: apps/site/pages/docs/customizing/mixins-breakpoints.mdx [hunk 10]
```diff
@@ -0,0 +1,246 @@
...
+
+
+<Tabs items={['Global mixins', 'Layout & Spacing Mixins', 'Focus Ring Mixins']}>
```
Comment (hellofanny) [inline apps/site/pages/docs/customizing/mixins-breakpoints.mdx:91]:
```suggestion
<Tabs items={['Global Mixins', 'Layout & Spacing Mixins', 'Focus Ring Mixins']}>
```

File: apps/site/pages/docs/customizing/mixins-breakpoints.mdx [hunk 11]
```diff
@@ -0,0 +1,246 @@
...
+
+        @include media(">=notebook") { padding-bottom: $padding; }
+      }
+    }
+    ```
+  </Tab>
+  <Tab>
+    ```scss copy
+    @mixin input-focus-ring($outline: #{var(--fs-color-focus-ring)}, $border: #{var(--fs-border-color-active)}) {
+      @media not all and (min-resolution: .001dpcm) { // Target only Safari browsers
```
Comment (hellofanny) [inline apps/site/pages/docs/customizing/mixins-breakpoints.mdx:156]: Actually, I think this is outdated: `Target only safari browser code`. We need to verify.

File: apps/site/pages/docs/customizing/mixins-breakpoints.mdx [hunk 12]
```diff
@@ -0,0 +1,246 @@
...
+          &:focus { @include focus-ring; }            // due to Safari's lack of support
+        }
+      }
+      &:focus-visible { @include focus-ring; }
+    }
+    ```
+  </Tab>
+</Tabs>
+
+### Custom Mixins
```
Comment (hellofanny) [inline apps/site/pages/docs/customizing/mixins-breakpoints.mdx:198]:
```suggestion
### Customizing Mixins
```

File: apps/site/pages/docs/customizing/mixins-breakpoints.mdx [hunk 13]
```diff
@@ -0,0 +1,246 @@
...
+
+The process of overriding those mixins is really similar to the custom breakpoints steps:
+
+1. Go to the `/src` folder and create the `styles` folder and inside it. You can run the following command to create them:
```
Comment (hellofanny) [inline apps/site/pages/docs/customizing/mixins-breakpoints.mdx:202]:
```suggestion
1. Go to the `src` folder and create the `styles` folder inside it. You can run the following command to create it:
```

File: apps/site/pages/docs/customizing/mixins-breakpoints.mdx [hunk 14]
```diff
@@ -0,0 +1,246 @@
...
+      }
+      &:focus-visible { @include focus-ring; }
+    }
+    ```
+  </Tab>
+</Tabs>
+
+### Custom Mixins
+
+The process of overriding those mixins is really similar to the custom breakpoints steps:
```
Comment (hellofanny) [inline apps/site/pages/docs/customizing/mixins-breakpoints.mdx:200]:
```suggestion
The process of overriding those mixins is similar to the customizing breakpoints steps. If you already have a `custom-mixin.scss` file, you can skip to Step 3.
```

File: apps/site/pages/docs/customizing/mixins-breakpoints.mdx [hunk 15]
```diff
@@ -0,0 +1,246 @@
...
+    ```
+  </Tab>
+  <Tab>
+    ```bash copy
+    touch src\styles\custom-mixins.scss
+    ```
+  </Tab>
+</Tabs>
+
+3. Choose the mixin you want to override and add it to the file:
```
Comment (hellofanny) [inline apps/site/pages/docs/customizing/mixins-breakpoints.mdx:232]:
```suggestion
3. Choose the [mixin](#mixins) you want to override and add it to the file:
```

File: apps/site/pages/docs/customizing/mixins-breakpoints.mdx [hunk 16]
```diff
@@ -0,0 +1,246 @@
...
+@import "@faststore/ui/utilities"
+
+$breakpoints: (
+  "phone":         320px,
+  "phonemid":      375px,
+  "tablet":        768px,
+  "notebook":      1280px,
+  "desktop":       1500px,
+);
+```
```
Comment (hellofanny) [inline apps/site/pages/docs/customizing/mixins-breakpoints.mdx:76]:
```suggestion
@import "@faststore/ui/utilities"

// Overwrites Breakpoints & Mixins ////////////////////

$breakpoints: (
  "phone":         320px,
  "phonemid":      375px,
  "tablet":        768px,
  "notebook":      1280px,
  "desktop":       1500px,
);
```
```

File: apps/site/pages/docs/customizing/mixins-breakpoints.mdx [hunk 17]
```diff
@@ -0,0 +1,246 @@
...
+  <Tab>
+    ```bash copy
+    touch src\styles\custom-mixins.scss
+    ```
+  </Tab>
+</Tabs>
+
+3. Choose the mixin you want to override and add it to the file:
+
+```diff copy
```
Comment (hellofanny) [inline apps/site/pages/docs/customizing/mixins-breakpoints.mdx:234]: I think this is a little bit confusing. 
How about first, we show how to add the mixin in this file, and then the "override changes applied"?

File: apps/site/pages/docs/customizing/mixins-breakpoints.mdx (hunk 15 above)
Comment (hellofanny) [inline apps/site/pages/docs/customizing/mixins-breakpoints.mdx:232]:
```suggestion
3. Choose the [mixin](#mixins) you want to override and add it to the file:

```scss {8} filename="src/styles/custom-mixins.scss" copy
@import "@faststore/ui/utilities"

// Overwrites Breakpoints & Mixins ////////////////////

@mixin input-focus-ring($outline: #{var(--fs-color-focus-ring-danger)}, $border: #{--fs-color-danger-border}) {
  [...]
}

```

File: packages/cli/src/utils/directory.ts [hunk 18]
```diff
@@ -21,10 +21,18 @@ export const srcFolderName = 'src'
 export const userSrcDir = path.join(userDir, srcFolderName)
 
 export const customizationsFolderName = 'customizations'
```
Comment (hellofanny) [inline packages/cli/src/utils/directory.ts:23]: Maybe we can just ignore this file change 🙈  Just indentation 🤔

Comment (renatamottam) [reply packages/cli/src/utils/directory.ts:23]: My bad 🤡

File: apps/site/pages/docs/customizing/mixins-breakpoints.mdx [hunk 19]
```diff
@@ -0,0 +1,260 @@
...
+  <Tab>
+    ```bash copy
+    touch src\styles\custom-mixins.scss
+    ```
+  </Tab>
+</Tabs>
+
+3. Copy and paste the following code snippet into the file:
+
+```scss {8} filename="src/styles/custom-mixins.scss" copy
```
Comment (hellofanny) [inline apps/site/pages/docs/customizing/mixins-breakpoints.mdx:66]:
```suggestion
```scss {10} filename="src/styles/custom-mixins.scss" copy
```

File: apps/site/pages/docs/customizing/mixins-breakpoints.mdx [hunk 20]
```diff
@@ -0,0 +1,260 @@
...
+@import "@faststore/ui/utilities"
+
+// Overwrites Breakpoints & Mixins ////////////////////
+
+@mixin input-focus-ring($outline: #{var(--fs-color-focus-ring-danger)}, $border: #{--fs-color-danger-border}) {
+  [...]
+}
+```
+
+4. Make the changes on the `@mixin` declaration.
```
Comment (hellofanny) [inline apps/site/pages/docs/customizing/mixins-breakpoints.mdx:246]:
```suggestion
This code will make the following changes on the default `@mixin input-focus-ring`:
```

File: apps/site/pages/docs/customizing/mixins-breakpoints.mdx [hunk 21]
```diff
@@ -0,0 +1,260 @@
...
+
+```diff copy
+- @mixin input-focus-ring($outline: #{var(--fs-color-focus-ring)}, $border: #{var(--fs-border-color-active)}) {
++ @mixin input-focus-ring($outline: #{var(--fs-color-focus-ring-danger)}, $border: #{--fs-color-danger-border}) {
+  [...]
+}
+```
+
+5. Run `yarn dev` in the terminal to start the development server. You should see that all components that use the `focus-ring` are now with a red shadow. Click the input to see the change.
```
Comment (hellofanny) [inline apps/site/pages/docs/customizing/mixins-breakpoints.mdx:255]:
```suggestion
5. Run `yarn dev` to start the development server. You should see that all components that use the `focus-ring` now have a red shadow when focused. Click on the input to see the change.
```

File: apps/site/pages/docs/customizing/mixins-breakpoints.mdx [hunk 22]
```diff
@@ -0,0 +1,260 @@
...
+);
+```
+
+<Callout type="warning" emoji="⚠️">
+  You need to declare all the variants (phone, phonemid, tablet, notebook and desktop) even when changing only one of the values.
+</Callout>
+
+4. Run `yarn dev` in the terminal to start the development server. Your should be able to see the changes.
+Modifying these breakpoints implies changes to certain global tokens. See [Grid & Layout](/docs/global-tokens/grid-and-layout#breakpoints) for more details.
+---
```
Comment (hellofanny) [inline apps/site/pages/docs/customizing/mixins-breakpoints.mdx:86]:
```suggestion


Modifying these breakpoints implies changes to certain global tokens. See [Grid & Layout](/docs/global-tokens/grid-and-layout#breakpoints) for more details.

---
```

**Comment by hahnwhitney** [comment]
I'm not an official reviewer, but this looks ready to merge from my side 🎉 

The docs explaining how to use the mixins are super helpful, and much more clear 😄

File: apps/site/pages/docs/customizing/mixins-breakpoints.mdx [hunk 23]
```diff
@@ -0,0 +1,267 @@
...
+  "desktop":       1440px,
+);
+```
+
+### Customizing Breakpoints
+
+Follow the steps below to override the pre-defined sizing system for the screen breakpoints.
+
+<Callout type="warning" emoji="⚠️">
+  Modifying these breakpoints implies changes to certain global tokens. See [Grid & Layout](/docs/global- tokens/grid-and-layout#breakpoints) for more details.
```
Comment (hellofanny) [inline apps/site/pages/docs/customizing/mixins-breakpoints.mdx:35]:
```suggestion
  Modifying these breakpoints implies changes to certain global tokens. See [Grid & Layout](/docs/global-tokens/grid-and-layout#breakpoints) for more details.
```

**Review by hellofanny — APPROVED** [review]

**Review by Mariana-Caetano — APPROVED** [review]

File: packages/core/src/styles/global/index.scss [hunk 24]
```diff
@@ -1,7 +1,10 @@
-/* establish a layer order up-front, FS UI base styles has the lowest priority */
+// Establish a layer order up-front, FS UI base styles have the lowest priority
 @layer base, components, theme;
 
-// Sections
+// FastStore UI's base styles
+@import "@faststore/ui/src/styles/global.scss";
+
+// Components & Sections styles
 @import "@faststore/ui/src/components/atoms/Overlay/styles.scss";
```
Comment (hellofanny) [inline packages/core/src/styles/global/index.scss:8]:
```suggestion
@import "@faststore/ui/src/components/atoms/Skeleton/styles.scss";
@import "@faststore/ui/src/components/atoms/Overlay/styles.scss";
```

**Pull request merged**